src/tevatron/datasets/dataset.py

```diff
@@ -0,0 +1,102 @@
+from datasets import load_dataset
+from .preprocessor import TrainPreProcessor, QueryPreProcessor, CorpusPreProcessor
+from ..arguments import DataArguments
+
+DEFAULT_PROCESSORS = [TrainPreProcessor, QueryPreProcessor, CorpusPreProcessor]
+PROCESSOR_INFO = {
+    'Tevatron/wikipedia-nq': DEFAULT_PROCESSORS,
+    'Tevatron/wikipedia-trivia': DEFAULT_PROCESSORS,
+    'Tevatron/wikipedia-curated': DEFAULT_PROCESSORS,
+    'Tevatron/wikipedia-wq': DEFAULT_PROCESSORS,
+    'Tevatron/wikipedia-squad': DEFAULT_PROCESSORS,
+    'Tevatron/scifact': DEFAULT_PROCESSORS,
+    'Tevatron/msmarco-passage': DEFAULT_PROCESSORS,
+    'json': [None, None, None]
+}
+
+
+class HFTrainDataset:
+    def __init__(self, tokenizer, data_args: DataArguments, cache_dir):
+        data_files = data_args.train_path
+        if data_files:
+            data_files = {data_args.dataset_split: data_files}
+        self.dataset = load_dataset(data_args.dataset_name,
+                                    data_args.dataset_language,
+                                    data_files=data_files, cache_dir=cache_dir)[data_args.dataset_split]
+        self.preprocessor = PROCESSOR_INFO[data_args.dataset_name][0] if data_args.dataset_name in PROCESSOR_INFO\
+            else DEFAULT_PROCESSORS[0]
+        self.tokenizer = tokenizer
+        self.q_max_len = data_args.q_max_len
+        self.p_max_len = data_args.p_max_len
+        self.proc_num = data_args.dataset_proc_num
+        self.neg_num = data_args.train_n_passages - 1
+        self.separator = data_args.passage_field_separator
```

Comment:
i feel like keeping `self.separator = getattr(tokenizer, data_args.passage_field_separator, data_args.passage_field_separator)` here will make it easy for users

Reply:
yeah, sure.